types/src/shared/utils/string_utils.ts

```diff
@@ -69,3 +69,14 @@ export function redactString(str: string, n: number) {
   const redacted = "*".repeat(str.length - n) + str.slice(-n);
   return redacted;
 }
+
+/**
+ * Returns an ellipsis if the string is too long, so it's never longer than
+ * `maxStringLength`.
+ * @param str The string to truncate.
+ */
+export function ellipsis(str: string, maxStringLength: number) {
```

Comment:
Why not using CSS classes for this?

Reply:
We could but IMHO the downsides are:
- not usable elswhere than inside tags in .tsx;
- less convenient, harder to setup ( text-ellipsis cannot be used alone, we need to think about other concerns such as do I want overflow-hidden, wrap/no wrap, do I need to set a max-width --> asking those questions is sometimes relevant, sometimes not and we lose time and complexify css just for the sake of the ellipsis)

Reply:
> less convenient, harder to setup ( text-ellipsis cannot be used alone, we need to think about other concerns such as do I want overflow-hidden, wrap/no wrap, do I need to set a max-width --> asking those questions is sometimes relevant, sometimes not and we lose time and complexify css just for the sake of the ellipsis)

To me, it can be configure as easily as what's being done here and offer a nicer UX. Here 20 characters might work on a laptop but not on mobile. Leveraging CSS covers everything 😉

Reply:
fine, fine :+1: